simulation/traffic_simulator/src/entity/ego_entity.cpp

```diff
@@ -273,37 +133,11 @@ auto EgoEntity::getRouteLanelets() const -> std::vector<std::int64_t>
   return ids;
 }
 
-auto EgoEntity::getCurrentPose() const -> geometry_msgs::msg::Pose
-{
-  Eigen::VectorXd relative_position(3);
-  relative_position(0) = vehicle_model_ptr_->getX();
-  relative_position(1) = vehicle_model_ptr_->getY();
-  relative_position(2) = 0.0;
-  relative_position =
-    quaternion_operation::getRotationMatrix(initial_pose_->orientation) * relative_position;
-
-  geometry_msgs::msg::Pose current_pose;
-  current_pose.position.x = initial_pose_.value().position.x + relative_position(0);
-  current_pose.position.y = initial_pose_.value().position.y + relative_position(1);
-  current_pose.position.z = initial_pose_.value().position.z + relative_position(2);
-  current_pose.orientation = [this]() {
-    geometry_msgs::msg::Vector3 rpy;
-    rpy.x = 0;
-    rpy.y = 0;
-    rpy.z = vehicle_model_ptr_->getYaw();
-    return initial_pose_.value().orientation *
-           quaternion_operation::convertEulerAngleToQuaternion(rpy);
-  }();
-
-  return current_pose;
-}
+auto EgoEntity::getCurrentPose() const -> geometry_msgs::msg::Pose { return status_.pose; }
 
 auto EgoEntity::getCurrentTwist() const -> geometry_msgs::msg::Twist
 {
-  geometry_msgs::msg::Twist current_twist;
-  current_twist.linear.x = vehicle_model_ptr_->getVx();
-  current_twist.angular.z = vehicle_model_ptr_->getWz();
-  return current_twist;
+  return status_.action_status.twist;
 }
```

Comment:
This implementation seems, same as EntityBase::getCurrentTwist()

Reply:
I've removed EgoEntity::getCurrentTwist() method.